quickwit/quickwit-config/src/storage_config.rs

```diff
@@ -370,11 +371,14 @@ impl S3StorageConfig {
     }
 
     pub fn force_path_style_access(&self) -> Option<bool> {
-        let force_path_style_access = get_bool_from_env(
-            "QW_S3_FORCE_PATH_STYLE_ACCESS",
-            self.force_path_style_access,
-        );
-        Some(force_path_style_access)
+        static FORCE_PATH_STYLE: OnceLock<Option<bool>> = OnceLock::new();
```

Comment:
Have you discovered if all that logging was justified? Is this normal that we created that many S3Storage?

Reply:
It was not unreasonable, but could be improved with caching, which is covered by https://github.com/quickwit-oss/quickwit/pull/5377 (not sure if it completely resolves the issue)